hypertrace-core-graphql-log-event-schema/src/main/java/org/hypertrace/core/graphql/log/event/dao/GatewayServiceLogEventsResponseConverter.java

```diff
@@ -0,0 +1,64 @@
+package org.hypertrace.core.graphql.log.event.dao;
+
+import io.reactivex.rxjava3.core.Observable;
+import io.reactivex.rxjava3.core.Single;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import javax.inject.Inject;
+import lombok.experimental.Accessors;
+import org.hypertrace.core.graphql.common.request.AttributeRequest;
+import org.hypertrace.core.graphql.common.utils.BiConverter;
+import org.hypertrace.core.graphql.log.event.request.LogEventRequest;
+import org.hypertrace.core.graphql.log.event.schema.LogEvent;
+import org.hypertrace.core.graphql.log.event.schema.LogEventResultSet;
+import org.hypertrace.gateway.service.v1.common.Value;
+import org.hypertrace.gateway.service.v1.log.events.LogEventsResponse;
+
+public class GatewayServiceLogEventsResponseConverter {
+
+  private final BiConverter<Collection<AttributeRequest>, Map<String, Value>, Map<String, Object>>
+      attributeMapConverter;
+
+  @Inject
+  GatewayServiceLogEventsResponseConverter(
+      BiConverter<Collection<AttributeRequest>, Map<String, Value>, Map<String, Object>>
+          attributeMapConverter) {
+    this.attributeMapConverter = attributeMapConverter;
+  }
+
+  public Single<LogEventResultSet> convert(LogEventRequest request, LogEventsResponse response) {
+    return Observable.fromIterable(response.getLogEventsList())
+        .flatMapSingle(logEvent -> this.convert(request, logEvent))
```

Comment:
nit: `concatMapSingle`. we want order to be preserved and `flatMapSingle` does that when single threaded, but we want to declare those semantics since we can't guarantee the threading here (the caller could change it). This issue came up after a lot of these implementations were written, so older code that's being referenced may still be using flatMap.

Reply:
Addressed.

> but we want to declare those semantics since we can't guarantee the threading here (the caller could change it). 

How can the caller change it?

Reply:
You can change the scheduler via subscribeOn/observeOn operators: http://reactivex.io/documentation/operators/subscribeon.html